doc/developer_guide.md

```diff
@@ -1,3 +1,62 @@
+# Environment setup
+
+You need Python >= 3.6 installed, and you need [tox](https://pypi.org/project/tox/) >= 3.10.
+
+Using tox takes care of all the details of creating and managing an isolated
+virtual environment, installing the necessary dependencies, and isolating
+testsuite runs.
+
+The command for using tox is: `tox {-e envlist} {-- arguments}`. Use `tox -a
+-v` to get a list of available environments.
+
+The same virtual environment is reused by the following tox environments:
+- `tox -e text -- ARGS`: run the testsuite. This is the default environment
+  when not precised.
+- `tox -e setup -- COMMAND`: run `./setup.py COMMAND`.
+- `tox -e launch -- ARGS`: run `./setup.py launch ARGS`.
+- `tox -e packaging_checks`: run the same packaging checks as the CI (add `--
+  -n` to see a dry-run of the exact checks).
+- `tox -e plugins_install`: install the distribution plugins (or the specified
+  plugins when run with `tox -e plugins_install -- REQS`). Note that this does
+  not use the plugins manager for installing.
+- `tox -e release_prepare -- NEW_VERSION`: execute all the steps necessary for
+  preparing a new release; patch the version to `NEW_VERSION`  and update
+  `NEWS.md`, staging all the changes for review.
+- `tox -e release_finalize`: finalize the release; commit the staged changes,
+  create an annotated tag, and print the git command necessary for pushing the
+  release to GitHub.
+
+The actual virtual environment lives in `.tox/dev`, and can be "activated" like
+any other virtual environment.
```

Comment:
I think that we should provide a link here to explain what `"activated" like any other virtual environment` means, maybe like this?
```suggestion
The actual virtual environment lives in `.tox/dev`, and can be "activated" like
any other [virtual environment.](https://docs.python.org/3/tutorial/venv.html)
```

Reply:
I'll link to https://virtualenv.pypa.io/en/latest/user_guide.html?highlight=activate#activators.